packages/amplify_auth_cognito/ios/Classes/FlutterFetchCognitoSessionResult.swift

```diff
@@ -30,70 +31,145 @@ struct FlutterFetchCognitoSessionResult {
+            }
+        }
       }
     }
     return sub
   }
 
-  func getIdentityId(session: AuthSession) throws -> String {
-    var id: String = ""
+  func getIdentityId(session: AuthSession) throws -> String? {
+    let userPoolOnly = isUPOnly(session: session)
+    var id: String? = nil
+    if (userPoolOnly) {
+      return id
+    }
     if let identityProvider = session as? AuthCognitoIdentityProvider {
       do {
         id = try identityProvider.getIdentityId().get()
       } catch {
-         throw error as! AuthError
+        throw error as! AuthError
       }
     }
     return id
   }
 
-  func getCredentials(session: AuthSession) throws -> [String: String] {
+func getCredentials(session: AuthSession) throws -> [String: String] {
     var credentialMap: [String: String] = [:]
-
+    let userPoolOnly = isUPOnly(session: session)
+    if (userPoolOnly) {
+        return credentialMap
+    }
     if let awsCredentialsProvider = session as? AuthAWSCredentialsProvider {
-      let creds =  try awsCredentialsProvider.getAWSCredentials().get()
+      do {
+        let creds =  try awsCredentialsProvider.getAWSCredentials().get()
         credentialMap["awsAccessKey"] = creds.accessKey
         credentialMap["awsSecretKey"] = creds.secretKey
+      } catch {
+        throw error as! AuthError
+      }
+      do {
+        let tempCreds = try awsCredentialsProvider.getAWSCredentials().get() as? AuthAWSTemporaryCredentials
+          if ((tempCreds?.sessionKey) != nil) {
+            credentialMap["sessionToken"] = tempCreds?.sessionKey
+          }
+      } catch {
+        print("Unable to cast session token.")
+      }
     }
-    
     return credentialMap;
   }
 
   func getTokens(session: AuthSession) throws -> [String: String] {
     var tokenMap: [String: String] = [:]
 
     if let cognitoTokenProvider = session as? AuthCognitoTokensProvider {
-      let tokens = try cognitoTokenProvider.getCognitoTokens().get()
-        tokenMap["accessToken"] = tokens.accessToken
-        tokenMap["idToken"] = tokens.idToken
-        tokenMap["refreshToken"] = tokens.refreshToken
+        do {
+            let tokens = try cognitoTokenProvider.getCognitoTokens().get()
+            tokenMap["accessToken"] = tokens.accessToken
+            tokenMap["idToken"] = tokens.idToken
+            tokenMap["refreshToken"] = tokens.refreshToken
+        } catch {
+            tokenMap["error"] = "You are currently signed out."
+        }
     }
     return tokenMap;
   }
+
+  func isUPOnly(session: AuthSession) -> Bool {
+    var isUserPoolOnly = false;
+    do {
+      let identityProvider = session as! AuthCognitoIdentityProvider
+      try identityProvider.getIdentityId().get()
+    } catch {
+      if case .service(_, _, let errorType) = error as? AuthError {
+        if case .invalidAccountTypeException = errorType as? AWSCognitoAuthError {
+          isUserPoolOnly = true
+        }
+      }
+    }
+    return isUserPoolOnly
+  }
```

Comment:
The logic is not very clear here from the code an I couldn't understand exactly what `isUserPoolOnly` meant. Documentation here will be helpful and unit tests.

Reply:
I'm adding comments and iOS unit tests.

Reply:
Is this a documented behavior in iOS that won't change in the future?

Reply:
`.invalidAccountTypeException` is also returned if you try to fetch auth session in signedOut state while unauth access is not enabled. Since this is a public error this should not change, but it is not documented. Here are the cases where we throw `invalidAccountTypeException`:

1. Fetching identity id while signedOut with unauth not enabled
1. Fetching AWS credentials while signedOut with unauth not enabled
1. Fetching usersub while signedIn to Cognito Identity Pool
1. Fetching cognito user pool token while signedIn to Cognito Identity Pool
1. Fetching identityId while signedIn to Cognito User pool but CIDP is not configured
1. Fetching AWS credentials while signedIn to Cognito User pool but CIDP is not configured

Reply:
I have changed the function name to something more descriptive (but very verbose), and it now checks to see if the user does not have an identity but IS signed in, which would indicate that they've only signed in with a user pool:

```
  // Checks if the user is signed in to a user pool without a CIDP
  func signedInToUserPoolWithoutCIDP(session: AuthSession) -> Bool {
    var userpoolSignInOnly = false;
    do {
      let identityProvider = session as! AuthCognitoIdentityProvider
      try identityProvider.getIdentityId().get()
    } catch {
      if case .service(_, _, let errorType) = error as? AuthError {
        if case .invalidAccountTypeException = errorType as? AWSCognitoAuthError {
            if (session.isSignedIn) {
                userpoolSignInOnly = true
            }
        }
      }
    }
    return userpoolSignInOnly
  }
```